src/components/Search/Search.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { EventHandler, FormEventHandler, useState } from "react";
+type IntrinsicElements = JSX.IntrinsicElements["input"];
+
+interface Props extends IntrinsicElements {
+  variation?: SearchVariation;
+  onSearch?: EventHandler<any>;
+  onInput?: FormEventHandler<HTMLInputElement>;
+  disabled?: boolean;
+  initialValue?: string;
+  placeholder?: string;
+  labelText?: string;
+}
+type SearchVariation = "default" | "big" | "small";
+
+/**
+ * Search Component
+ * @param {string}            variation             Renders the style of the search component.
+ * @param {EventHandler}      onSearch              Handles its behavior when the search button is clicked or the user presses enter in the input. Returns value of the input. Returns value of the input.
+ * @param {EventHandler}      onInput               Handles its behavior when the user is typing.
+ * @param {boolean}           [disabled]            Determines whether or not a button is enabled.
+ * @param {string}            [initialValue]        If provided, used as value of search input.
+ * @param {string}            [placeholder]         The text placeholder for this component.
+ * @param {string}            [labelText]           Label text for screen reader.
+ */
+
+export const Search: React.FC<Props> = ({
+  variation = "default",
+  disabled = false,
+  onSearch,
```

Comment:
```suggestion
  onSearch = () => {} ,
```
If you don't want to make `onSearch()` required, it could also default to an empty function and the other changes could still work.